### PR TITLE
Make win32.mak compatible with MODEL=32omf

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
The previous assumption was `MODEL=32` because `MODEL=32mscoff` is built by `win64.mak`.

Verified locally because Azure requires further changes outside of this repository.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
This will need to be force-merged because it requires additional changes to `dmd/windows.sh` and `phobos/win32.mak`

**Comment by RazvanN7** [comment]
@MoonlightSentinel Ping me when you need this merged.

**Comment by MoonlightSentinel** [comment]
Submitted the required changes to `windows.sh` here: dlang/dmd#13572. That should enable Azure to build Druntime with `MODEL=32omf` again (but still fail on Phobos - will open another PR for that).

**Comment by MoonlightSentinel** [comment]
Azure successfully built druntime! Just one patch left... (Phobos/win32.mak)

**Comment by MoonlightSentinel** [comment]
Can you temporarily mark the Azure job as not required? Or do we have to modify the configuration until the Phobos PR is merged?

**Comment by RazvanN7** [comment]
I don't know how to do that and I might not have the rights, however, I can force merge this if that is required.

**Comment by MoonlightSentinel** [comment]
It'll be required - neither makefile update is sufficient on its own (this fails due to the outdated `phobos/win32.mak` and a Phobos PR cannot build Druntime without this patch)

**Comment by RazvanN7** [comment]
@MoonlightSentinel Thanks for fixing this!

**Pull request merged**
